test/katex-spec.js

```diff
@@ -1042,6 +1042,16 @@ describe("A left/right parser", function() {
         var normalEmpty = "\\Bigl .";
         expect(normalEmpty).toParse();
     });
+
+    it("should handle \\middle", function() {
+        var normalMiddle = "\\left( \\dfrac{x}{y} \\middle| \\dfrac{y}{z} \\right)";
+        expect(normalMiddle).toParse();
+    });
+
+    it("should error when \\middle is not in \\left...\\right", function() {
+        var unmatchedMiddle = "(\\middle|\\dfrac{x}{y})";
```

Comment:
Ah but it *does* work. I know it shouldn't, but catching this error case would be harder, and I didn't see much point.